### DISCUSSION TITLE
[Q] Subtitles?

### DISCUSSION THREAD

**Discussion by uxandai** [itsToggle/plex_debrid#114] [Q&A]
Hi there.

Any ideas how to integrate plex_debrid with some kind of automatic subtitle downloader?
Because plex_debrid approach is brilliant, but there's no way to use bazarr.
Also, I can't watch (monitor) rd downloads folder and download subtitles with external software and use it with plex.

PS Sub-Zero is still working?

**Comment by uxandai** [comment]
This is promising: https://github.com/replicatingTrouts/plex-sub-downloader

**Reply by itsToggle** [reply]
I usually use the subtitle search that's integrated into plex. When playing any media item, you can go to the "subtitles" menu and search opensubtitles.org and similar services for subtitles for your version of the movie/episode :)

**Reply by uxandai** [reply]
@itsToggle yeah, it works for 90% of the time, but sometimes opensubitles doesn't have matched subs.

**Reply by itsToggle** [reply]
Okay so I thought about this a bit further and there is sadly not much that can be done:

- You cant simply add files to realdebrid, so any subtitle downloader cant place downloaded subtitles into the correct folder.
- I could integrate a setting to only download releases that contain .srt (or similar subtile format) files, but that would exclude movie releases where the subtitles are placed within the .mkv/.mp4 container. It would also decrease the chances of finding a cached release, since sometimes only the mkv/mp4 file is cached on realdebrid and not the subtitle files.
- There is the subtitle finder within plex I mentioned before, but as you said, its pure luck wether or not there is a subtitle that is correctly timed to the particular release you download.

That seems to be about it.. sorry :/

**Reply by uxandai** [reply]
@itsToggle you're 90% correct. I've tested plex-sub-downloader and it places subtitles directly in Plex db, not file system ;)

**Reply by sasagr** [reply]
> This is promising: https://github.com/replicatingTrouts/plex-sub-downloader

Hi. I had the same question and I m afraid Sub-Zero is not working, like also the opensubtitles plex default plugin is not working if we are using the Plex default agent (required for plex-debrid to work properly). I tried this tool that you posted and it seems very interesting and basically covering my needs but I m having some troubles to make it working. Is it working with you? I opened an issue on its github

**Comment by xplorer990** [comment]
Perhaps "Sub Zero" is an option. I haven't tested it yet but thought to share it.
https://github.com/pannal/Sub-Zero.bundle

My plex came preinstalled with "plex web tools" which I think you need for sub zero to work.

https://github.com/ukdtom/WebTools.bundle

**Reply by sasagr** [reply]
Sub zero is not working with the new Plex agents. Moreover it is a plugin not in develop anymore

**Comment by AndyZaa** [comment]
I have plex-sub-downloader installed and working nicely, but it doesn't really address most of my issue with subtitle management in plex_debrid. I get a lot of subtitle tracks of SSA/ASS and PGS which cause Plex to perform some intense transcoding. I can't use the SSA/ASS or PGS subtitles on those videos. SRT subs work fine. 

plex-sub-downloader only works automatically on newly watchlisted vidoes. plex-sub-downloader won't add SRT or any other subs to the Plex database if it sees that there are already any type of existing subs. Deleting the incompatible subs would clear the way to download SRT subs with plex-sub-downloader, at least using the individual metadata key method. However I haven't found any method to manage and delete subs in the Plex database. 

Pasta, https://github.com/cglatot/pasta, comes close but doesn't allow deletion. It would be great if plex-sub-downloader allowed upload of SRT subs to videos that already have subs, or if it provided for deletion of existing subs while uploading SRT subs. As well, if plex-sub-downloader allowed updating videos already in the database (not just new) it would be a total solution.
